src/axom/core/examples/core_containers.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2017-2020, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*! \file core_containers.cpp
+ *  \brief This example code is a demonstration of the Axom Core containers.
+ */
+
+ /* This example code contains snippets used in the Core Sphinx documentation.
+  * They begin and end with comments such as
+  *
+  * timer_start
+  * timer_end
+  *
+  * each prepended with an underscore.
+  */
+
+
+// Axom includes
+#include "axom/core/Array.hpp"
+#include "axom/core/memory_management.hpp"
+
+#ifdef WIN32
+#include "windows.h"
+void sleep(int numSeconds)
+{
+  int numMilliSecs = numSeconds * 1000;
+  Sleep(numMilliSecs);
+}
+#else
+#include <unistd.h> // for sleep()
+#endif
+
+// C/C++ includes
+#include <iostream>
+#include <vector>
+
+void showArray(axom::Array<int> &a, const char* name)
```

Comment:
Minor: Any reason not to use `std::string` here instead of `const char*`?

Reply:
Ease and variety of use in all kinds of code.